File: frontend/lib/models/wallet.dart
class Wallet {
  final double balance;
  final String currency;
  final double lockedAmount;

  Wallet({required this.balance, this.currency = 'USD', this.lockedAmount = 0.0});

  factory Wallet.fromJson(Map<String, dynamic> json) => Wallet(
    balance: (json['balance'] as num).toDouble(),
    currency: json['currency'] ?? 'USD',
    lockedAmount: (json['locked_amount'] as num?)?.toDouble() ?? 0.0,
  );
}

class WalletTransaction {
  final int id;
  final String type;
  final double amount;
  final String reference;
  final String description;
  final String createdAt;

  WalletTransaction({required this.id, required this.type, required this.amount,
                     required this.reference, required this.description, required this.createdAt});

  factory WalletTransaction.fromJson(Map<String, dynamic> json) => WalletTransaction(
    id: json['id'], type: json['type'], amount: (json['amount'] as num).toDouble(),
    reference: json['reference'], description: json['description'],
    createdAt: json['created_at'],
  );
}

File: backend/app/models/wallet.py
from datetime import datetime
from sqlalchemy import String, Float, DateTime, func, ForeignKey, Text, Enum as SAEnum
from sqlalchemy.orm import Mapped, mapped_column, relationship
import enum

from .base import Base


class TransactionType(str, enum.Enum):
    CREDIT = "credit"
    DEBIT = "debit"


class Wallet(Base):
    __tablename__ = "wallets"

    id: Mapped[int] = mapped_column(primary_key=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id"), unique=True)
    balance: Mapped[float] = mapped_column(Float, default=0.0)
    currency: Mapped[str] = mapped_column(String(3), default="USD")
    locked_amount: Mapped[float] = mapped_column(Float, default=0.0)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

    user = relationship("User")


class WalletTransaction(Base):
    __tablename__ = "wallet_transactions"

    id: Mapped[int] = mapped_column(primary_key=True)
    wallet_id: Mapped[int] = mapped_column(ForeignKey("wallets.id"))
    type: Mapped[TransactionType] = mapped_column(SAEnum(TransactionType))
    amount: Mapped[float] = mapped_column(Float)
    reference: Mapped[str] = mapped_column(String(255))
    description: Mapped[str] = mapped_column(Text, default="")
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())

    wallet = relationship("Wallet")
